AF-2094: Test GWT - AppFormer JS editor bridge classes

jesuino opened the pull request:
Tests for the envelope GWT classes.

jesuino committed:
AF-2094: Test GWT - AppFormer JS editor bridge classes

tiagobento commented on packages/microeditor-envelope/src/__tests__/AppFormerGwtApi.test.ts:
```diff
@@ -0,0 +1,33 @@
+const dummyEditor: DummyEditor = new DummyEditor();
+const dummyGWTEditor: DummyGWTEditor = new DummyGWTEditor(dummyEditor);
+const editorId:string = "dummy editor";
+const appFormerGwtApi: AppFormerGwtApi = new AppFormerGwtApi();
```
I think you could remove the explicit types on these declarations, since they can all be inferred

tiagobento commented on packages/microeditor-envelope/src/__tests__/AppFormerGwtApi.test.ts:
```diff
@@ -0,0 +1,33 @@
+  public get(): AppFormer.Editor {
+    return this.wrappedEditor;
+  }
+}
```
Are you using the Prettier formatter? I think the default is to have a blank line between methods..

tiagobento commented on packages/microeditor-envelope/src/__tests__/AppFormerGwtApi.test.ts:
```diff
@@ -0,0 +1,33 @@
+import * as AppFormer from "appformer-js-core";
+import { GwtEditor } from '../gwt/GwtEditor';
+
+class DummyGWTEditor {
```
We should go for camel-case acronyms in class names too, so this class should be `DummyGwtEditor`.

tiagobento replied:
Also on the variables

tiagobento commented on packages/microeditor-envelope/src/__tests__/AppFormerGwtApi.test.ts:
```diff
@@ -0,0 +1,33 @@
+    expect(possibleEditor).toBeTruthy();
+  });
+  test("get non existing editor", () => {
+    expect(() => appFormerGwtApi.getEditor("X")).toThrowError("GwtEditor with id 'X' was not found");
```
I think that just checking for an error is enough, we don't want to duplicate the error message on the tests.

tiagobento commented on packages/microeditor-envelope/src/__tests__/GwtEditorWrapper.test.ts:
```diff
@@ -0,0 +1,33 @@
+    const removedListBarHeader = document.querySelector(".panel-heading.uf-listbar-panel-header");
+    expect(removedHeaderPanel).toBeFalsy();
+    expect(removedListBarHeader).toBeFalsy();
+  });
```
I think you could add a test that checks for a no-op behaviour when these elements don't exist as well.. It's a good thing to know that a method that's not supposed to fail, doesn't...

tiagobento replied:
Also, on this test, I think it's a good thing to improve readability to vertically space things apart. To make it clearer I would wrap `wrapper.af_onOpen()` aroud blank lines so that it is clear what is the test preparation, what is the method we're testing and what are the assertions.

tiagobento commented on packages/microeditor-envelope/src/__tests__/GwtEditorWrapperFactory.test.ts:
```diff
@@ -0,0 +1,53 @@
+    window.appFormerGwtFinishedLoading();
+    await delay(100);
+    expect(res).toBeCalled();
+  });
```
Same comment here for blank lines to logically separate the phases of the test

tiagobento reviewed (changes requested):
Looks good in general, just added some comments inline. Just make sure that you're using Prettier to format the code and that there's no redundant type declarations! Thanks @jesuino!

jesuino commented:
Addressed these issues on https://github.com/tiagobento/kogito-tooling/pull/1

Closing this, we can continue our discussion in https://github.com/tiagobento/kogito-tooling/pull/1
